### PR TITLE
rgw:multisite: fix RGWRadosRemoveOmapKeysCR and change cn to intrusive_ptr

### PR DESCRIPTION
**Pull request by shashalu**
RGWRadosRemoveOmapKeysCR::request_complete return val is wrong. The return val should get from completion.  Some member variables is not used, clear up those variables.
change raw 'RGWAioCompletionNotifier*' to intrusive_ptr for rgw_cr_rados

Fixes:http://tracker.ceph.com/issues/20539

Signed-off-by: Shasha Lu <lu.shasha@eisoo.com>

### CODE REVIEW

**Comment by shashalu** [comment]
@cbodley  @yuriw  Sorry, forget `cn->get();` in send_request, which will result in crash in request_complete.  Revised it.

File: src/rgw/rgw_cr_rados.cc [hunk 1]
```diff
@@ -283,6 +283,7 @@ int RGWRadosGetOmapKeysCR::send_request() {
   op.omap_get_vals2(marker, max_entries, entries, nullptr, &rval);
 
   cn = stack->create_completion_notifier();
+  cn->get();
```
Comment (cbodley) [inline src/rgw/rgw_cr_rados.cc]: was this meant to go in `RGWRadosRemoveOmapKeysCR::send_request()`?

it looks like this reference on `cn` is missing from a lot of these coroutines. could you please go through and change these raw `RGWAioCompletionNotifier*` pointers to `boost::intrusive_ptr<RGWAioCompletionNotifier>`? see `RGWRadosRemoveCR` as an example

**Comment by yuriw** [comment]
un-tagged

 http://pulpito.ceph.com/yuriw-2017-07-08_02:56:51-rgw-wip-yuri-testing2_2017_7_9-distro-basic-smithi/

**Comment by shashalu** [comment]
@cbodley  already revised.

File: src/rgw/rgw_cr_rados.cc [hunk 2]
```diff
@@ -222,9 +222,6 @@ RGWRadosSetOmapKeysCR::RGWRadosSetOmapKeysCR(RGWRados *_store,
 
 RGWRadosSetOmapKeysCR::~RGWRadosSetOmapKeysCR()
 {
-  if (cn) {
-    cn->put();
-  }
```
Comment (cbodley) [inline src/rgw/rgw_cr_rados.cc]: thanks. can you remove these destructors now that they're empty? (`~RGWRadosTimelogAddCR()` and `~RGWRadosTimelogTrimCR()` also)

**Comment by cbodley** [comment]
tested the version with intrusive_ptr under valgrind and found no leaks 👍

**Comment by shashalu** [comment]
@cbodley  updated

**Pull request merged**